download-markdown/download-markdown.js

```diff
@@ -0,0 +1,172 @@
+import path from 'path';
+import fs from 'fs';
+import { fetch, timeoutSignal, AbortError } from '@adobe/fetch';
+import { saveToFile, entryToPath } from '../bulk-update/document-manager/document-manager.js';
+import { localizeStageUrl } from '../bulk-update/bulk-update.js';
+
+const delay = (milliseconds) => new Promise((resolve) => { setTimeout(resolve, milliseconds); });
+
+const ALLOW_SKIP = true; // Allow skipping files that already exist
+const PAGE_DELAY = 500; // 500ms delay for fetching from hlx.page
+const LIVE_DELAY = 0; // 0ms delay for fetching from live site
+
+/**
+ * Reads a JSON file from the specified directory.
+ * @param {string} file - The name of the JSON file.
+ * @param {string} directory - The directory where the file is located.
+ * @returns {object} - The parsed JSON object.
+ */
+function readJsonFile(file, directory) {
+  const filePath = path.join(directory, file);
+  if (!fs.existsSync(filePath)) {
+    console.error(`File not found: ${filePath}`);
+    return null;
+  }
+  return JSON.parse(fs.readFileSync(filePath, 'utf8'));
+}
+
+/**
+ * Fetches markdown content from the specified URL.
+ *
+ * @param {string} url - The URL of the markdown file to fetch.
+ * @param {number} fetchWaitMs - The delay in milliseconds before making the fetch request.
+ * @param {function} fetchFn - The fetch function to use.
+ * @returns {Promise<string>} A promise that resolves to the fetched markdown content as a string.
+ */
+export async function fetchMarkdown(url, fetchWaitMs, fetchFn = fetch) {
+  try {
+    console.log(`Fetching markdown ${url}, delay ${fetchWaitMs}ms, timeout 5s`);
+    await delay(fetchWaitMs); // Wait 500ms to avoid rate limiting, not needed for live.
+    const signal = timeoutSignal(5000); // 5s timeout
+    const response = await fetchFn(url, { signal });
+
+    if (!response.ok) {
+      console.warn('Failed to fetch markdown.', response.status, response.statusText);
+      signal.clear();
+      return '';
+    }
+    const text = await response.text();
+    signal.clear();
+    return text;
+  } catch (e) {
+    if (e instanceof AbortError) {
+      console.warn('Fetch timed out after 1s');
```

Comment:
should update to `Fetch timed out after 5s`? due to signal length?

Reply:
Should be 5s or 5000ms, I'll set the timeout to a constant and use that here and the above console log.